src/graph/visitor/test/ExtractGroupSuiteVisitorTest.cpp

```diff
@@ -0,0 +1,380 @@
+/* Copyright (c) 2022 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License.
+ */
+
+#include <gtest/gtest.h>
+
+#include "graph/context/ast/CypherAstContext.h"
+#include "graph/visitor/ExtractGroupSuiteVisitor.h"
+#include "graph/visitor/test/VisitorTestBase.h"
+
+namespace nebula {
+namespace graph {
+class ExtractGroupSuiteVisitorTest : public VisitorTestBase {
+ public:
+  bool check(GroupSuite g1, GroupSuite g2) {
+    if (g1.groupKeys.size() != g2.groupKeys.size() ||
+        g1.groupItems.size() != g2.groupItems.size()) {
+      std::cout << "g1.groupKeys.size()=" << g1.groupKeys.size() << std::endl;
+      std::cout << "g2.groupKeys.size()=" << g2.groupKeys.size() << std::endl;
+      std::cout << "g1.groupItems.size()=" << g1.groupItems.size() << std::endl;
+      std::cout << "g2.groupItems.size()=" << g2.groupItems.size() << std::endl;
+      return false;
+    }
+    for (unsigned int i = 0; i < g1.groupKeys.size(); i++) {
+      if (g1.groupKeys[i]->toString() != g2.groupKeys[i]->toString()) {
+        std::cout << "g1.groupKeys[i]=" << g1.groupKeys[i]->toString() << std::endl;
+        std::cout << "g2.groupKeys[i]=" << g2.groupKeys[i]->toString() << std::endl;
+        return false;
+      }
+    }
+    for (unsigned int i = 0; i < g1.groupItems.size(); i++) {
+      if (g1.groupItems[i]->toString() != g2.groupItems[i]->toString()) {
+        std::cout << "g1.groupItems[i]=" << g1.groupItems[i]->toString() << std::endl;
+        std::cout << "g2.groupItems[i]=" << g2.groupItems[i]->toString() << std::endl;
+        return false;
+      }
+    }
+    return true;
+  }
+};
+
+TEST_F(ExtractGroupSuiteVisitorTest, TestConstantExpression) {
```

Comment:
You should test some nested expression scenarios.

Reply:
Yes. I do. TEST_F(ExtractGroupSuiteVisitorTest, TestUnaryExpression2)

Reply:
Add test case:
`(abs(v.age)+1)+count(v.name)`